### PR TITLE
New row crash

### PR DESCRIPTION
**Pull request by spireken**

### CODE REVIEW

File: Applications/Spire/Source/KeyBindings/KeyBindingItemDelegate.cpp [hunk 1]
```diff
@@ -59,8 +54,15 @@ void KeyBindingItemDelegate::on_editing_finished() {
 bool KeyBindingItemDelegate::eventFilter(QObject* watched, QEvent* event) {
   if(event->type() == QEvent::KeyPress) {
     auto e = static_cast<QKeyEvent*>(event);
-    if(e->key() == Qt::Key_Tab || e->key() == Qt::Key_Backtab) {
-      m_key_signal(static_cast<Qt::Key>(e->key()));
+    if(e->key() == Qt::Key_Tab) {
```
Comment (andrei-sa) [inline Applications/Spire/Source/KeyBindings/KeyBindingItemDelegate.cpp:57]: Keep `if(... || ...)`, then branch on `if(e->key() == Qt::Key_Tab)` for `closeEditor`.

File: Applications/Spire/Source/KeyBindings/KeyBindingsTableView.cpp [hunk 2]
```diff
@@ -345,35 +357,20 @@ void KeyBindingsTableView::on_column_selection_changed(
...
-    if(new_index.flags().testFlag(Qt::ItemIsEditable)) {
-      m_is_editing_cell = true;
-      m_table->edit(new_index);
-    }
-    update();
+void KeyBindingsTableView::on_editor_closed(const QWidget* editor,
+    QAbstractItemDelegate::EndEditHint hint) {
+  if(hint == QAbstractItemDelegate::EditNextItem) {
+    auto next = get_next_editable_index(m_table->currentIndex());
+    edit_cell(next);
```
Comment (andrei-sa) [inline Applications/Spire/Source/KeyBindings/KeyBindingsTableView.cpp:370]: Move `edit_cell` out. Use conditional lambda to initialize next/prev if necessary.

**Comment by spireken** [comment]
> Looks fine for the most part. I assume the fix itself is functional as tested by others.

Yeah, sorry, the tester is called KeyBindingsUiTester-NewRowCrash. It was OK'd by Darryl and Fayaz.

File: Applications/Spire/Source/KeyBindings/KeyBindingsTableView.cpp [hunk 3]
```diff
@@ -200,14 +198,15 @@ bool KeyBindingsTableView::eventFilter(QObject* watched, QEvent* event) {
       }();
       m_is_default_cell_selected = false;
       m_is_editing_cell = false;
+      auto editor_open = m_table->isPersistentEditorOpen(
```
Comment (kranar) [inline Applications/Spire/Source/KeyBindings/KeyBindingsTableView.cpp:201]: Booleans are named ```is_xxx```.

**Pull request merged**